mmdet/core/anchor/point_generator.py

```diff
@@ -35,3 +37,199 @@ def valid_flags(self, featmap_size, valid_size, device='cuda'):
         valid_xx, valid_yy = self._meshgrid(valid_x, valid_y)
         valid = valid_xx & valid_yy
         return valid
+
+
+@PRIOR_GENERATORS.register_module()
+class MlvlPointGenerator:
+    """Standard points generator for multi-level(Mlvl) feature maps in 2D
+    points-based detectors.
```

Comment:
Suggest to merge PointGenerator and MlvlPointGenerator. 
From[here](https://github.com/open-mmlab/mmdetection/blob/670ecc25460ae583ccabb0b4f0182a284e17d0cc/mmdet/models/dense_heads/reppoints_head.py#L95) it seems RepPoints essentially use multi-level PointGenerator. So at least we can replace that.
To prevent BC-breaking, we can add an option named `return_list=False` by default.

Reply:
The original `PointGenerator` actually is a `single_level_points_genertator`, the original `grid_point` in `PointGenerator`  actually is `single_level_grid_priors` of `MlvlPointGenerator`, It is hard to merge them because the original implementation does not follow the design of `AnchorGenerator`.
I believe the best way is to replace `PointGenerator`  with `MlvlPointGenerator`, but I am afraid of BC-breaking.